### PR TITLE
Organise courses into categories

### PR DESCRIPTION
**Pull request by vunn-1010**
- Change database so that a courses belong to one category
- View courses by category

Layout
image

[Ticket Redmine](https://edu-redmine.sun-asterisk.vn/issues/24794)

### CODE REVIEW

**Comment by vunn-1010** [comment]
ready

File: app/Repositories/CategoryRepository.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+        return $categories;
+    }
+
+    /**
+     * Get one category by given id
+     */
+    public function find($id)
+    {
+        $category = Category::find($id);
```
Comment (quoctd-0607) [inline app/Repositories/CategoryRepository.php:36]: xử lý trường hợp không tìm thấy category

File: app/Repositories/CategoryRepository.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        return $category;
+    }
+
+    /**
+     * Get one category with its courses by given id
+     */
+    public function findWithCourses($id)
+    {
+        $category = Category::with('courses')
+            ->find($id);
```
Comment (quoctd-0607) [inline app/Repositories/CategoryRepository.php:47]: tương tự

**Comment by vunn-1010** [comment]
ready

**Pull request merged**